tests/test_nbr_health.py

```diff
@@ -27,7 +27,7 @@ def check_snmp(hostname, mgmt_addr, localhost, community, is_eos):
 
 def check_eos_facts(hostname, mgmt_addr, host):
     logger.info("Check neighbor {} eos facts".format(hostname))
-    res = host.eos_facts()
+    res = host.eos_facts(gather_subset=["!config"])
```

Comment:
just question here, what it means by ["!config"]? @vivekverma-arista

Reply:
Hi @StormLiangMS

If you see here https://github.com/sonic-net/sonic-mgmt/blob/master/tests/test_nbr_health.py#L42, the test is trying to access `ansible_net_interfaces` later from this collection.

If we just run 
```
res = host.eos_facts()
```

then that key is not present in the collection, therefore hitting a key error later
```
(Pdb) res1=host.eos_facts()
Thursday 18 April 2024  10:12:30 +0000 (0:00:36.672)       0:01:09.951 ******** 
[WARNING]: ansible-pylibssh not installed, falling back to paramiko
(Pdb) 'ansible_net_interfaces' in res1['ansible_facts']
False
```

on the other hand

```
(Pdb) res2 = host.eos_facts(gather_subset=["!config"])
Thursday 18 April 2024  10:14:29 +0000 (0:01:58.409)       0:03:08.360 ******** 
[WARNING]: ansible-pylibssh not installed, falling back to paramiko
(Pdb) 'ansible_net_interfaces' in res2['ansible_facts']
True
```

The reason is that there was a change to `arista.eos` https://github.com/ansible-collections/arista.eos/pull/343/files.
Prior to this PR the default behaviour itself was to return  `["!config"]` which today is `[min]`, therefore with the current ansible version in sonic-mgmt (which is 2.13.13)  we need to pass `["!config"]` explicitly.